editor/src/messages/portfolio/document/graph_operation/utility_types.rs

```diff
@@ -114,70 +123,115 @@ impl<'a> ModifyInputsContext<'a> {
 		Some(new_id)
 	}
 
-	pub fn skip_artboards(&self, output: &mut NodeOutput) -> Option<(NodeId, usize)> {
-		while let NodeInput::Node { node_id, output_index, .. } = &self.document_network.nodes.get(&output.node_id)?.inputs[output.node_output_index] {
-			let sibling_node = self.document_network.nodes.get(node_id)?;
-			if sibling_node.name != "Artboard" {
-				return Some((*node_id, *output_index));
+	/// Starts at any folder, or the output, and skips layer nodes based on insert_index. Non layer nodes are always skipped. Returns the post node id, pre node id, and the input index
+	///			-> Post node input_index: 0
+	///		↑		if skip_layer_nodes == 0, return (Post node, Some(Layer1), 1)
+	///	->	Layer1	input_index: 1
+	///		↑		if skip_layer_nodes == 1, return (Layer1, Some(Layer2), 0)
+	///	->	Layer2	input_index: 2
+	///		↑
+	///	->	NLN
+	///		↑		if skip_layer_nodes == 2, return (NLN, Some(Layer3), 0)
+	///	->	Layer3	input_index: 3
+	///				if skip_layer_nodes == 3, return (Layer3, None, 0)
```

Comment:
Please don't use tab characters in ASCII art. Can you please fix up the formatting for this so it's nice and readable?